### ISSUE TITLE
Gradle 5.2 warning on Gradle 6.0 incompatibility

### ISSUE THREAD

**Issue by elizarov** [JetBrains/gradle-idea-ext-plugin#69]
The following `build.gradle.kts` script reproduces the problem with Gradle 5.2:

```
plugins {
    id("org.jetbrains.gradle.plugin.idea-ext") version "0.5"
}

repositories {
    mavenCentral()
}
```

Run `gradlew model --warning-mode all` and get:

```
> Configure project :
Internal API constructor DefaultPolymorphicDomainObjectContainer(Class<T>, Instantiator) has been deprecated. This is scheduled to be removed in Gradle 6.0.
```

**Comment by bric3** [comment]
The same appear on non kotlin build script.

**Comment by bric3** [comment]
I'm not sure this can be changed easily without factoring out this internal class or without having a new version that targets gradle 5 code base. The following code

https://github.com/JetBrains/gradle-idea-ext-plugin/blob/41da16652d7e15665d22804143c3b7155d3f487c/src/main/groovy/org/jetbrains/gradle/ext/RunConfigurations.groovy#L23-L29

The problematic deprecation is the 2 argument form of the constructor, however in Gradle 4, the other form have 3 arguments : 

https://github.com/gradle/gradle/blob/v4.10.3/subprojects/core/src/main/java/org/gradle/api/internal/DefaultPolymorphicDomainObjectContainer.java#L30-L37

While Gradle 5 version non deprecated constructors use 4 arguments:

https://github.com/gradle/gradle/blob/v5.3.0/subprojects/core/src/main/java/org/gradle/api/internal/DefaultPolymorphicDomainObjectContainer.java#L31-L47

Note I'm a groovy neophyte, I barely know the language and features.

----------

However this part of the code could be invoked dynamically depending on the gradle version 

https://github.com/JetBrains/gradle-idea-ext-plugin/blob/41da16652d7e15665d22804143c3b7155d3f487c/src/main/groovy/org/jetbrains/gradle/ext/GradleUtils.groovy#L17-L20

**Comment by AlexLandau** [comment]
It looks like this is fixed in releases 0.6.1 and 0.7.
